compaction/compaction_manager.cc

```diff
@@ -2128,9 +2128,6 @@ compaction_backlog_tracker::compaction_backlog_tracker(compaction_backlog_tracke
 compaction_backlog_tracker&
 compaction_backlog_tracker::operator=(compaction_backlog_tracker&& x) noexcept {
     if (this != &x) {
-        if (auto manager = std::exchange(_manager, x._manager)) {
```

Comment:
compaction_backlog_tracker::compaction_backlog_tracker(compaction_backlog_tracker&& other) should be fixed the same way. if other is registered, we should allow it to be deregistered naturally via its ctor.  as in move assignment, the problem is benign in practice

Reply:
What do you mean?
This hunk deregisters `this` from `this->_manager`, but in the move ctor `this->_manager` will always be null.

Reply:
I opened https://github.com/scylladb/scylladb/issues/15248
I include a fix for it in the next revision